src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnectionBuilderExtensions.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.SignalR.Internal.Protocol;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.SignalR.Client
+{
+    public static class HubConnectionBuilderExtensions
+    {
+        public static IHubConnectionBuilder WithHubProtocol(this IHubConnectionBuilder hubConnectionBuilder, IHubProtocol hubProtocol)
+        {
+            hubConnectionBuilder.AddSetting(HubConnectionBuilderDefaults.HubProtocolKey, hubProtocol);
+            return hubConnectionBuilder;
+        }
+
+        public static IHubConnectionBuilder WithLogger(this IHubConnectionBuilder hubConnectionBuilder, ILoggerFactory loggerFactory)
```

Comment:
Should we add WithConsoleLogger?

Reply:
Added WithConsoleLogger. Skipped loggingBuilder for now (I assume it's useful mainly when running the client on the server?)

Reply:
No actually it'll be useful in a bunch of scenarios (like application insights on the client side).

Reply:
And with Generic Host, it will become possible for a client application to have a LoggingBuilder

Reply:
I am going to merge what we have and do LoggingBuilder separately. I am not sure if it's worth squeezing to alpha1 - this is the first time we are talking about it - we can decide once the PR is out.

Reply:
I think we should wait with ILoggingBuilder until [this](https://github.com/aspnet/Logging/issues/679) is done

Reply:
https://github.com/aspnet/SignalR/issues/832